src/GDGUkraine/utils.py

```diff
@@ -114,3 +122,283 @@ def make_vcard(user_reg, url=None):
+                logger.debug('It is class-based routed app')
+                logger.debug(script)
+                logger.debug(app)
+                urls.update(retrieve_class_routes(app.root,
+                                                  mp=app.script_name))
+            elif isinstance(request_dispatcher,
+                            cp._cpdispatch.RoutesDispatcher):
+                logger.debug('It is Routes routed app')
+                logger.debug('Skipping...')
+                logger.debug(script)
+                logger.debug(app)
+                logger.debug(request_dispatcher)
+                for handler_name in request_dispatcher.controllers.keys():
+                    if handler_name in urls['__routes__']:
+                        logger.warn('Handler name `{}` is already in routes '
+                                    'URL map. Avoid having same identifier for'
+                                    ' different paths!'.format(handler_name))
+                    urls['__routes__'][handler_name] = {
+                        'script': app.script_name,
+                        'mapper': request_dispatcher.mapper}
+        url_resolve_map = urls
+        return urls
+    else:
+        return urls
+
+
+def url_for_class(handler, url_args=[], url_params={}):
+    app_name = __name__.split('.')[0].lower()
+    handler = handler.lower()
+
+    if handler.split('.')[0] != app_name:
+        handler = '.'.join([app_name, handler])
+
+    url_route = url_resolve_map.get(handler)
+    logger.debug(url_route)
+    return cp.url(uri_builder(url_route, *url_args, **url_params),
+                  script_name='',
+                  base=base_url())
+
+
+def url_for_routes(handler, **url_params):
+    try:
+        routes_map = url_resolve_map['__routes__']
+        _ = routes_map[handler]
+        mapper = _['mapper']
+        script_name = _['script']
+
+        old_mapper = None
+        if hasattr(routes.request_config(), 'mapper'):
+            old_mapper = routes.request_config().mapper
+
+        # When running tests it's empty
+        # But standalone run results in WSGI env stored,
+        # which has 'SCRIPT_NAME'. Hacking routes to avoid prepending it
+        old_environ = None
+        if hasattr(routes.request_config(), 'environ'):
+            old_environ = routes.request_config().environ.copy()
+            routes.request_config().environ.clear()
+
+        old_prefix = None
+        if hasattr(routes.request_config(), 'prefix'):
+            old_prefix = routes.request_config().prefix
+
+        routes.request_config().mapper = mapper
+        routes_url = routes.url_for(handler, **url_params)
+
+        # Restore everything
+        if old_mapper:
+            routes.request_config().mapper = old_mapper
+
+        if old_environ:
+            routes.request_config().environ = old_environ
+
+        if old_prefix:
+            routes.request_config().prefix = old_prefix
+    except KeyError:
```

Comment:
Probably 
```python
except KeyError as ke:
    raise TypeError(...) from ke
```

Will give a little bit clearer traceback:
But it is just my opinion. You might or might not agree w/ it =)